<details open>
<summary>French version</summary>

# Data collection and preparation

Ce notebook met en place la base de donnees de l'analyse. Le code commence par importer les librairies utiles, puis interroge l'API ouverte de **StatsBomb** afin d'identifier la competition cible et de recuperer les matchs de la Coupe du Monde 2022.

La logique generale est progressive: partir d'un catalogue de competitions, isoler la bonne saison, recuperer la liste complete des matchs, puis descendre vers les matchs des deux finalistes et enfin vers les evenements detailles de la finale. Cette progression est importante, car elle garantit que les notebooks suivants s'appuient sur un perimetre de donnees propre et coherent.

</details>

<details>
<summary>English version</summary>

# Data Collection and Preparation

This notebook sets up the database behind the analysis. The code starts by importing the useful libraries, then queries the open **StatsBomb** API in order to identify the target competition and retrieve the matches from the 2022 World Cup.

The overall logic is progressive: start from a competition catalog, isolate the correct season, retrieve the full list of matches, then move down to the matches played by the two finalists, and finally to the detailed events from the final itself. This progression matters because it ensures that the following notebooks are built on a clean and coherent data perimeter.

</details>


In [ ]:
from statsbombpy import sb
import pandas as pd
import os
pd.set_option("display.max_columns", None)
pd.set_option("display.width", 1000)

print("Toutes les librairies importées avec succès !")

Toutes les librairies importées avec succès !


In [3]:
competitions = sb.competitions()

wc = competitions[competitions["competition_name"] == "FIFA World Cup"]

wc[["competition_id", "season_id", "season_name"]]


c:\Users\HP\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsbombpy\api_client.py:21: NoAuthWarning: credentials were not supplied. open data access only
  warnings.warn(


,competition_id,season_id,season_name
29,43,106,2022
30,43,3,2018
31,43,55,1990
32,43,54,1986
33,43,51,1974
34,43,272,1970
35,43,270,1962
36,43,269,1958


<details open>
<summary>French version</summary>

## Filtrer les matchs de la France et de l'Argentine

Ici, le code cree deux sous-ensembles a partir de la table globale des **64 matchs** du tournoi: un pour la **France** et un pour l'**Argentine**. Le filtrage repose sur les colonnes `home_team` et `away_team`, ce qui permet de recuperer tous les matchs joues par chaque equipe, quel que soit son statut dans la rencontre.

Les sorties montrent bien le parcours complet des deux selections, avec **7 matchs** chacune entre la phase de groupes et la finale du **18 decembre 2022**. Cette verification est utile, car elle confirme que la collecte couvre l'integralite du parcours des finalistes.

On remarque aussi que certaines cellules de collecte sont repetees plus loin dans le notebook. Cela ne change pas le fond de l'analyse: les memes objets sont reconstruits, puis reutilises pour la suite.

</details>

<details>
<summary>English version</summary>

## Filter France and Argentina Matches

Here, the code creates two subsets from the global table of the tournament's **64 matches**: one for **France** and one for **Argentina**. The filtering uses the `home_team` and `away_team` columns, which makes it possible to retrieve every match played by each team, regardless of home or away status.

The outputs clearly show the full route of both national teams, with **7 matches** each between the group stage and the final on **18 December 2022**. This check is useful because it confirms that the data collection covers the entire route of the finalists.

It is also worth noting that some collection cells are repeated later in the notebook. This does not change the substance of the analysis: the same objects are rebuilt and then reused afterwards.

</details>


In [4]:
matches = sb.matches(competition_id=43, season_id=106)

print(f"{len(matches)} matchs trouvés")

matches[["match_id", "match_date", "home_team", "away_team", "home_score", "away_score"]].head(10)

c:\Users\HP\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsbombpy\api_client.py:21: NoAuthWarning: credentials were not supplied. open data access only
  warnings.warn(


64 matchs trouvés


,match_id,match_date,home_team,away_team,home_score,away_score
0,3857256,2022-12-02,Serbia,Switzerland,2,3
1,3869151,2022-12-03,Argentina,Australia,2,1
2,3857257,2022-11-30,Australia,Denmark,1,0
3,3857258,2022-11-24,Brazil,Serbia,2,0
4,3857288,2022-11-26,Tunisia,Australia,0,1
5,3857267,2022-11-29,Ecuador,Senegal,1,2
6,3869321,2022-12-09,Netherlands,Argentina,2,2
7,3857287,2022-11-24,Uruguay,South Korea,0,0
8,3869486,2022-12-10,Morocco,Portugal,1,0
9,3869685,2022-12-18,Argentina,France,3,3


In [5]:
matches


,match_id,match_date,kick_off,competition,season,home_team,away_team,home_score,away_score,match_status,match_status_360,last_updated,last_updated_360,match_week,competition_stage,stadium,referee,home_managers,away_managers,data_version,shot_fidelity_version,xy_fidelity_version
0,3857256,2022-12-02,21:00:00.000,International - FIFA World Cup,2022,Serbia,Switzerland,2,3,available,available,2023-02-17T23:45:15.306706,2023-04-26T23:49:58.956186,3,Group Stage,Stadium 974,Fernando Andrés Rapallini,Dragan Stojković,Murat Yakin,1.1.0,2,2
1,3869151,2022-12-03,21:00:00.000,International - FIFA World Cup,2022,Argentina,Australia,2,1,available,available,2023-07-30T07:46:05.382784,2023-07-30T07:48:51.865595,4,Round of 16,Ahmad bin Ali Stadium,Szymon Marciniak,Lionel Sebastián Scaloni,Graham James Arnold,1.1.0,2,2
2,3857257,2022-11-30,17:00:00.000,International - FIFA World Cup,2022,Australia,Denmark,1,0,available,available,2023-04-28T17:13:10.958725,2023-06-20T11:04:37.638969,3,Group Stage,Al Janoub Stadium,Mustapha Ghorbal,Graham James Arnold,Kasper Hjulmand,1.1.0,2,2
3,3857258,2022-11-24,21:00:00.000,International - FIFA World Cup,2022,Brazil,Serbia,2,0,available,available,2023-06-24T17:17:27.911026,2023-07-11T14:56:31.096588,1,Group Stage,Lusail Stadium,Alireza Faghani,Telê Santana da Silva,Dragan Stojković,1.1.0,2,2
4,3857288,2022-11-26,12:00:00.000,International - FIFA World Cup,2022,Tunisia,Australia,0,1,available,available,2023-02-28T21:25:20.888552,2023-04-27T00:30:07.835815,2,Group Stage,Al Janoub Stadium,Daniel Siebert,Jalel Kadri,Graham James Arnold,1.1.0,2,2
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
59,3857265,2022-11-22,18:00:00.000,International - FIFA World Cup,2022,Mexico,Poland,0,0,available,available,2023-04-19T06:53:07.045195,2023-04-27T00:49:30.005344,1,Group Stage,Stadium 974,Chris Beath,Gerardo Daniel Martino,Czesław Michniewicz,1.1.0,2,2
60,3857262,2022-12-02,17:00:00.000,International - FIFA World Cup,2022,South Korea,Portugal,2,1,available,available,2023-03-17T22:58:19.035038,2023-04-26T23:48:08.999798,3,Group Stage,Education City Stadium,Facundo Tello Figueroa,Paulo Jorge Gomes Bento,Fernando Manuel Fernandes da Costa Santos,1.1.0,2,2
61,3857261,2022-11-29,21:00:00.000,International - FIFA World Cup,2022,Wales,England,0,3,available,available,2023-03-27T07:45:00.892102,2023-04-27T00:02:44.241621,3,Group Stage,Ahmad bin Ali Stadium,Slavko Vinčić,Robert Page,Gareth Southgate,1.1.0,2,2
62,3857255,2022-12-01,21:00:00.000,International - FIFA World Cup,2022,Japan,Spain,2,1,available,available,2023-03-30T12:03:47.965683,2023-04-26T23:51:48.352990,3,Group Stage,Sheikh Khalifa International Stadium,Victor Miguel de Freitas Gomes,Hajime Moriyasu,Luis Enrique Martínez García,1.1.0,2,2


In [6]:
france_matches = matches[
    (matches["home_team"] == "France") | (matches["away_team"] == "France")
].copy()

arg_matches = matches[
    (matches["home_team"] == "Argentina") | (matches["away_team"] == "Argentina")
].copy()

print("Matchs de la France :")
france_matches[["match_date", "home_team", "away_team", "home_score", "away_score"]].sort_values("match_date")

Matchs de la France :


,match_date,home_team,away_team,home_score,away_score
49,2022-11-22,France,Australia,4,1
12,2022-11-26,France,Denmark,2,1
53,2022-11-30,Tunisia,France,1,0
26,2022-12-04,France,Poland,3,1
20,2022-12-10,England,France,1,2
21,2022-12-14,France,Morocco,2,0
9,2022-12-18,Argentina,France,3,3


In [7]:
print("Matchs de l'Argentine :")
arg_matches[["match_date", "home_team", "away_team", "home_score", "away_score"]].sort_values("match_date")

Matchs de l'Argentine :


,match_date,home_team,away_team,home_score,away_score
37,2022-11-22,Argentina,Saudi Arabia,1,2
13,2022-11-26,Argentina,Mexico,2,0
11,2022-11-30,Poland,Argentina,0,2
1,2022-12-03,Argentina,Australia,2,1
6,2022-12-09,Netherlands,Argentina,2,2
19,2022-12-13,Argentina,Croatia,3,0
9,2022-12-18,Argentina,France,3,3


In [8]:
pd.set_option("display.max_columns", None)
pd.set_option("display.width", 1000)

print("Toutes les librairies importées avec succès !")

Toutes les librairies importées avec succès !


In [9]:
competitions = sb.competitions()

wc = competitions[competitions["competition_name"] == "FIFA World Cup"]

wc[["competition_id", "season_id", "season_name"]]


c:\Users\HP\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsbombpy\api_client.py:21: NoAuthWarning: credentials were not supplied. open data access only
  warnings.warn(


,competition_id,season_id,season_name
29,43,106,2022
30,43,3,2018
31,43,55,1990
32,43,54,1986
33,43,51,1974
34,43,272,1970
35,43,270,1962
36,43,269,1958


In [10]:
matches = sb.matches(competition_id=43, season_id=106)

print(f"{len(matches)} matchs trouvés")

matches[["match_id", "match_date", "home_team", "away_team", "home_score", "away_score"]].head(10)

64 matchs trouvés


c:\Users\HP\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsbombpy\api_client.py:21: NoAuthWarning: credentials were not supplied. open data access only
  warnings.warn(


,match_id,match_date,home_team,away_team,home_score,away_score
0,3857256,2022-12-02,Serbia,Switzerland,2,3
1,3869151,2022-12-03,Argentina,Australia,2,1
2,3857257,2022-11-30,Australia,Denmark,1,0
3,3857258,2022-11-24,Brazil,Serbia,2,0
4,3857288,2022-11-26,Tunisia,Australia,0,1
5,3857267,2022-11-29,Ecuador,Senegal,1,2
6,3869321,2022-12-09,Netherlands,Argentina,2,2
7,3857287,2022-11-24,Uruguay,South Korea,0,0
8,3869486,2022-12-10,Morocco,Portugal,1,0
9,3869685,2022-12-18,Argentina,France,3,3


In [11]:
matches


,match_id,match_date,kick_off,competition,season,home_team,away_team,home_score,away_score,match_status,match_status_360,last_updated,last_updated_360,match_week,competition_stage,stadium,referee,home_managers,away_managers,data_version,shot_fidelity_version,xy_fidelity_version
0,3857256,2022-12-02,21:00:00.000,International - FIFA World Cup,2022,Serbia,Switzerland,2,3,available,available,2023-02-17T23:45:15.306706,2023-04-26T23:49:58.956186,3,Group Stage,Stadium 974,Fernando Andrés Rapallini,Dragan Stojković,Murat Yakin,1.1.0,2,2
1,3869151,2022-12-03,21:00:00.000,International - FIFA World Cup,2022,Argentina,Australia,2,1,available,available,2023-07-30T07:46:05.382784,2023-07-30T07:48:51.865595,4,Round of 16,Ahmad bin Ali Stadium,Szymon Marciniak,Lionel Sebastián Scaloni,Graham James Arnold,1.1.0,2,2
2,3857257,2022-11-30,17:00:00.000,International - FIFA World Cup,2022,Australia,Denmark,1,0,available,available,2023-04-28T17:13:10.958725,2023-06-20T11:04:37.638969,3,Group Stage,Al Janoub Stadium,Mustapha Ghorbal,Graham James Arnold,Kasper Hjulmand,1.1.0,2,2
3,3857258,2022-11-24,21:00:00.000,International - FIFA World Cup,2022,Brazil,Serbia,2,0,available,available,2023-06-24T17:17:27.911026,2023-07-11T14:56:31.096588,1,Group Stage,Lusail Stadium,Alireza Faghani,Telê Santana da Silva,Dragan Stojković,1.1.0,2,2
4,3857288,2022-11-26,12:00:00.000,International - FIFA World Cup,2022,Tunisia,Australia,0,1,available,available,2023-02-28T21:25:20.888552,2023-04-27T00:30:07.835815,2,Group Stage,Al Janoub Stadium,Daniel Siebert,Jalel Kadri,Graham James Arnold,1.1.0,2,2
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
59,3857265,2022-11-22,18:00:00.000,International - FIFA World Cup,2022,Mexico,Poland,0,0,available,available,2023-04-19T06:53:07.045195,2023-04-27T00:49:30.005344,1,Group Stage,Stadium 974,Chris Beath,Gerardo Daniel Martino,Czesław Michniewicz,1.1.0,2,2
60,3857262,2022-12-02,17:00:00.000,International - FIFA World Cup,2022,South Korea,Portugal,2,1,available,available,2023-03-17T22:58:19.035038,2023-04-26T23:48:08.999798,3,Group Stage,Education City Stadium,Facundo Tello Figueroa,Paulo Jorge Gomes Bento,Fernando Manuel Fernandes da Costa Santos,1.1.0,2,2
61,3857261,2022-11-29,21:00:00.000,International - FIFA World Cup,2022,Wales,England,0,3,available,available,2023-03-27T07:45:00.892102,2023-04-27T00:02:44.241621,3,Group Stage,Ahmad bin Ali Stadium,Slavko Vinčić,Robert Page,Gareth Southgate,1.1.0,2,2
62,3857255,2022-12-01,21:00:00.000,International - FIFA World Cup,2022,Japan,Spain,2,1,available,available,2023-03-30T12:03:47.965683,2023-04-26T23:51:48.352990,3,Group Stage,Sheikh Khalifa International Stadium,Victor Miguel de Freitas Gomes,Hajime Moriyasu,Luis Enrique Martínez García,1.1.0,2,2


In [12]:
france_matches = matches[
    (matches["home_team"] == "France") | (matches["away_team"] == "France")
].copy()

arg_matches = matches[
    (matches["home_team"] == "Argentina") | (matches["away_team"] == "Argentina")
].copy()

print("Matchs de la France :")
france_matches[["match_date", "home_team", "away_team", "home_score", "away_score"]].sort_values("match_date")

Matchs de la France :


,match_date,home_team,away_team,home_score,away_score
49,2022-11-22,France,Australia,4,1
12,2022-11-26,France,Denmark,2,1
53,2022-11-30,Tunisia,France,1,0
26,2022-12-04,France,Poland,3,1
20,2022-12-10,England,France,1,2
21,2022-12-14,France,Morocco,2,0
9,2022-12-18,Argentina,France,3,3


In [13]:
print("Matchs de l'Argentine :")
arg_matches[["match_date", "home_team", "away_team", "home_score", "away_score"]].sort_values("match_date")

Matchs de l'Argentine :


,match_date,home_team,away_team,home_score,away_score
37,2022-11-22,Argentina,Saudi Arabia,1,2
13,2022-11-26,Argentina,Mexico,2,0
11,2022-11-30,Poland,Argentina,0,2
1,2022-12-03,Argentina,Australia,2,1
6,2022-12-09,Netherlands,Argentina,2,2
19,2022-12-13,Argentina,Croatia,3,0
9,2022-12-18,Argentina,France,3,3


In [14]:
pd.set_option("display.max_columns", None)
pd.set_option("display.width", 1000)

print("Toutes les librairies importées avec succès !")

Toutes les librairies importées avec succès !


In [15]:
competitions = sb.competitions()

wc = competitions[competitions["competition_name"] == "FIFA World Cup"]

wc[["competition_id", "season_id", "season_name"]]


c:\Users\HP\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsbombpy\api_client.py:21: NoAuthWarning: credentials were not supplied. open data access only
  warnings.warn(


,competition_id,season_id,season_name
29,43,106,2022
30,43,3,2018
31,43,55,1990
32,43,54,1986
33,43,51,1974
34,43,272,1970
35,43,270,1962
36,43,269,1958


In [16]:
matches = sb.matches(competition_id=43, season_id=106)

print(f"{len(matches)} matchs trouvés")

matches[["match_id", "match_date", "home_team", "away_team", "home_score", "away_score"]].head(10)

64 matchs trouvés


c:\Users\HP\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsbombpy\api_client.py:21: NoAuthWarning: credentials were not supplied. open data access only
  warnings.warn(


,match_id,match_date,home_team,away_team,home_score,away_score
0,3857256,2022-12-02,Serbia,Switzerland,2,3
1,3869151,2022-12-03,Argentina,Australia,2,1
2,3857257,2022-11-30,Australia,Denmark,1,0
3,3857258,2022-11-24,Brazil,Serbia,2,0
4,3857288,2022-11-26,Tunisia,Australia,0,1
5,3857267,2022-11-29,Ecuador,Senegal,1,2
6,3869321,2022-12-09,Netherlands,Argentina,2,2
7,3857287,2022-11-24,Uruguay,South Korea,0,0
8,3869486,2022-12-10,Morocco,Portugal,1,0
9,3869685,2022-12-18,Argentina,France,3,3


In [17]:
matches


,match_id,match_date,kick_off,competition,season,home_team,away_team,home_score,away_score,match_status,match_status_360,last_updated,last_updated_360,match_week,competition_stage,stadium,referee,home_managers,away_managers,data_version,shot_fidelity_version,xy_fidelity_version
0,3857256,2022-12-02,21:00:00.000,International - FIFA World Cup,2022,Serbia,Switzerland,2,3,available,available,2023-02-17T23:45:15.306706,2023-04-26T23:49:58.956186,3,Group Stage,Stadium 974,Fernando Andrés Rapallini,Dragan Stojković,Murat Yakin,1.1.0,2,2
1,3869151,2022-12-03,21:00:00.000,International - FIFA World Cup,2022,Argentina,Australia,2,1,available,available,2023-07-30T07:46:05.382784,2023-07-30T07:48:51.865595,4,Round of 16,Ahmad bin Ali Stadium,Szymon Marciniak,Lionel Sebastián Scaloni,Graham James Arnold,1.1.0,2,2
2,3857257,2022-11-30,17:00:00.000,International - FIFA World Cup,2022,Australia,Denmark,1,0,available,available,2023-04-28T17:13:10.958725,2023-06-20T11:04:37.638969,3,Group Stage,Al Janoub Stadium,Mustapha Ghorbal,Graham James Arnold,Kasper Hjulmand,1.1.0,2,2
3,3857258,2022-11-24,21:00:00.000,International - FIFA World Cup,2022,Brazil,Serbia,2,0,available,available,2023-06-24T17:17:27.911026,2023-07-11T14:56:31.096588,1,Group Stage,Lusail Stadium,Alireza Faghani,Telê Santana da Silva,Dragan Stojković,1.1.0,2,2
4,3857288,2022-11-26,12:00:00.000,International - FIFA World Cup,2022,Tunisia,Australia,0,1,available,available,2023-02-28T21:25:20.888552,2023-04-27T00:30:07.835815,2,Group Stage,Al Janoub Stadium,Daniel Siebert,Jalel Kadri,Graham James Arnold,1.1.0,2,2
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
59,3857265,2022-11-22,18:00:00.000,International - FIFA World Cup,2022,Mexico,Poland,0,0,available,available,2023-04-19T06:53:07.045195,2023-04-27T00:49:30.005344,1,Group Stage,Stadium 974,Chris Beath,Gerardo Daniel Martino,Czesław Michniewicz,1.1.0,2,2
60,3857262,2022-12-02,17:00:00.000,International - FIFA World Cup,2022,South Korea,Portugal,2,1,available,available,2023-03-17T22:58:19.035038,2023-04-26T23:48:08.999798,3,Group Stage,Education City Stadium,Facundo Tello Figueroa,Paulo Jorge Gomes Bento,Fernando Manuel Fernandes da Costa Santos,1.1.0,2,2
61,3857261,2022-11-29,21:00:00.000,International - FIFA World Cup,2022,Wales,England,0,3,available,available,2023-03-27T07:45:00.892102,2023-04-27T00:02:44.241621,3,Group Stage,Ahmad bin Ali Stadium,Slavko Vinčić,Robert Page,Gareth Southgate,1.1.0,2,2
62,3857255,2022-12-01,21:00:00.000,International - FIFA World Cup,2022,Japan,Spain,2,1,available,available,2023-03-30T12:03:47.965683,2023-04-26T23:51:48.352990,3,Group Stage,Sheikh Khalifa International Stadium,Victor Miguel de Freitas Gomes,Hajime Moriyasu,Luis Enrique Martínez García,1.1.0,2,2


<details open>
<summary>French version</summary>

## Isoler la finale puis charger la table d'evenements

Le code identifie ensuite le match commun a la France et a l'Argentine afin d'isoler automatiquement la finale. Cette etape retourne un identifiant unique de match, ici **3869685**, qui sert ensuite a telecharger la table d'evenements associee avec `sb.events()`.

On obtient alors un jeu de donnees bien plus fin, compose de **4407 evenements**. A ce stade, on ne travaille plus seulement sur les scores ou les feuilles de match, mais sur la granularite complete du jeu: passes, controles, conduites de balle, duels, pressions, tirs, fautes ou recuperations.

</details>

<details>
<summary>English version</summary>

## Isolate the Final and Load the Event Table

The code then identifies the match shared by France and Argentina in order to isolate the final automatically. This step returns a unique match identifier, here **3869685**, which is then used to download the associated event table with `sb.events()`.

At that point, we obtain a much finer dataset made up of **4407 events**. From there, the work no longer focuses only on scores or match sheets, but on the full granularity of play: passes, controls, carries, duels, pressures, shots, fouls, or recoveries.

</details>


In [18]:
france_matches = matches[
    (matches["home_team"] == "France") | (matches["away_team"] == "France")
].copy()

arg_matches = matches[
    (matches["home_team"] == "Argentina") | (matches["away_team"] == "Argentina")
].copy()

print("Matchs de la France :")
france_matches[["match_date", "home_team", "away_team", "home_score", "away_score"]].sort_values("match_date")

Matchs de la France :


,match_date,home_team,away_team,home_score,away_score
49,2022-11-22,France,Australia,4,1
12,2022-11-26,France,Denmark,2,1
53,2022-11-30,Tunisia,France,1,0
26,2022-12-04,France,Poland,3,1
20,2022-12-10,England,France,1,2
21,2022-12-14,France,Morocco,2,0
9,2022-12-18,Argentina,France,3,3


In [19]:
print("Matchs de l'Argentine :")
arg_matches[["match_date", "home_team", "away_team", "home_score", "away_score"]].sort_values("match_date")

Matchs de l'Argentine :


,match_date,home_team,away_team,home_score,away_score
37,2022-11-22,Argentina,Saudi Arabia,1,2
13,2022-11-26,Argentina,Mexico,2,0
11,2022-11-30,Poland,Argentina,0,2
1,2022-12-03,Argentina,Australia,2,1
6,2022-12-09,Netherlands,Argentina,2,2
19,2022-12-13,Argentina,Croatia,3,0
9,2022-12-18,Argentina,France,3,3


<details open>
<summary>French version</summary>

## Comprendre la structure des evenements

Les comptages par type d'evenement servent de controle qualitatif sur les donnees chargees. Les categories les plus frequentes sont les **passes (1263)**, les **ball receipts (1114)** et les **carries (940)**, devant les **pressions (361)** puis les actions plus disruptives comme les duels, dribbles, fautes ou tirs.

Cette hierarchie est coherente avec la nature d'un match de football evenementiel: la circulation du ballon produit naturellement l'essentiel du volume. Le code permet donc ici de verifier que la table de la finale est exploitable et de reperer les variables les plus interessantes pour les notebooks suivants, en particulier pour l'etude des passes, des tirs et des sequences menant au danger offensif.

</details>

<details>
<summary>English version</summary>

## Understanding the Event Structure

The counts by event type serve as a qualitative control on the loaded data. The most frequent categories are **passes (1263)**, **ball receipts (1114)**, and **carries (940)**, ahead of **pressures (361)** and then more disruptive actions such as duels, dribbles, fouls, or shots.

This hierarchy is consistent with the nature of event data in football: ball circulation naturally produces most of the volume. The code therefore helps verify that the final-event table is usable and highlights the most relevant variables for the following notebooks, especially for studying passes, shots, and the sequences leading to attacking danger.

</details>


In [20]:
finale = matches[
    (matches["home_team"].isin(["France", "Argentina"])) &
    (matches["away_team"].isin(["France", "Argentina"]))
].copy()

finale[["match_id", "match_date", "home_team", "away_team", "home_score", "away_score"]]

,match_id,match_date,home_team,away_team,home_score,away_score
9,3869685,2022-12-18,Argentina,France,3,3


In [21]:
finale_id = finale["match_id"].values[0]
print(f"ID de la finale : {finale_id}")

ID de la finale : 3869685


In [22]:
events_finale = sb.events(match_id=finale_id)

print(f"Nombre d'événements : {len(events_finale)}")
events_finale.head()

c:\Users\HP\AppData\Local\Programs\Python\Python312\Lib\site-packages\statsbombpy\api_client.py:21: NoAuthWarning: credentials were not supplied. open data access only
  warnings.warn(


Nombre d'événements : 4407


,50_50,bad_behaviour_card,ball_receipt_outcome,ball_recovery_offensive,ball_recovery_recovery_failure,block_deflection,block_offensive,carry_end_location,clearance_aerial_won,clearance_body_part,clearance_head,clearance_left_foot,clearance_other,clearance_right_foot,counterpress,dribble_nutmeg,dribble_outcome,dribble_overrun,duel_outcome,duel_type,duration,foul_committed_advantage,foul_committed_card,foul_committed_offensive,foul_committed_penalty,foul_committed_type,foul_won_advantage,foul_won_defensive,foul_won_penalty,goalkeeper_body_part,goalkeeper_end_location,goalkeeper_outcome,goalkeeper_position,goalkeeper_technique,goalkeeper_type,id,index,interception_outcome,location,match_id,minute,off_camera,out,pass_aerial_won,pass_angle,pass_assisted_shot_id,pass_body_part,pass_cross,pass_deflected,pass_end_location,pass_goal_assist,pass_height,pass_inswinging,pass_length,pass_outcome,pass_outswinging,pass_recipient,pass_recipient_id,pass_shot_assist,pass_switch,pass_technique,pass_through_ball,pass_type,period,play_pattern,player,player_id,position,possession,possession_team,possession_team_id,related_events,second,shot_aerial_won,shot_body_part,shot_end_location,shot_first_time,shot_freeze_frame,shot_key_pass_id,shot_one_on_one,shot_outcome,shot_statsbomb_xg,shot_technique,shot_type,substitution_outcome,substitution_outcome_id,substitution_replacement,substitution_replacement_id,tactics,team,team_id,timestamp,type,under_pressure
0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0584ee21-e3dd-4d9f-95a0-5b5e84be25c3,1,NaN,NaN,3869685,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1,Regular Play,NaN,NaN,NaN,1,Argentina,779,NaN,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"{'formation': 433, 'lineup': [{'player': {'id'...",Argentina,779,00:00:00.000,Starting XI,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,b32679f8-942e-4122-96a2-015caf75e628,2,NaN,NaN,3869685,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1,Regular Play,NaN,NaN,NaN,1,Argentina,779,NaN,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"{'formation': 4231, 'lineup': [{'player': {'id...",France,771,00:00:00.000,Starting XI,NaN
2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,954f6855-de22-46a2-8d09-6fe94eec2b9b,3,NaN,NaN,3869685,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1,Regular Play,NaN,NaN,NaN,1,Argentina,779,[6404a8e8-afaf-489d-b65e-173a237ffed5],0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,France,771,00:00:00.000,Half Start,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6404a8e8-afaf-489d-b65e-173a237ffed5,4,NaN,NaN,3869685,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1,Regular Play,NaN,NaN,NaN,1,Argentina,779,[954f6855-de22-46a2-8d09-6fe94eec2b9b],0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Argentina,779,00:00:00.000,Half Start,NaN
4,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,e84fb5b3-270f-4427-bf97-1a6eaec7285f,1752,NaN,NaN,3869685,45,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2,From Free Kick,NaN,NaN,NaN,79,France,771,[3e8de8c9-8be8-4921-96ec-1a29ce8fd3d4],0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,France,771,00:00:00.000,Half Start,NaN


In [23]:
events_finale

,50_50,bad_behaviour_card,ball_receipt_outcome,ball_recovery_offensive,ball_recovery_recovery_failure,block_deflection,block_offensive,carry_end_location,clearance_aerial_won,clearance_body_part,clearance_head,clearance_left_foot,clearance_other,clearance_right_foot,counterpress,dribble_nutmeg,dribble_outcome,dribble_overrun,duel_outcome,duel_type,duration,foul_committed_advantage,foul_committed_card,foul_committed_offensive,foul_committed_penalty,foul_committed_type,foul_won_advantage,foul_won_defensive,foul_won_penalty,goalkeeper_body_part,goalkeeper_end_location,goalkeeper_outcome,goalkeeper_position,goalkeeper_technique,goalkeeper_type,id,index,interception_outcome,location,match_id,minute,off_camera,out,pass_aerial_won,pass_angle,pass_assisted_shot_id,pass_body_part,pass_cross,pass_deflected,pass_end_location,pass_goal_assist,pass_height,pass_inswinging,pass_length,pass_outcome,pass_outswinging,pass_recipient,pass_recipient_id,pass_shot_assist,pass_switch,pass_technique,pass_through_ball,pass_type,period,play_pattern,player,player_id,position,possession,possession_team,possession_team_id,related_events,second,shot_aerial_won,shot_body_part,shot_end_location,shot_first_time,shot_freeze_frame,shot_key_pass_id,shot_one_on_one,shot_outcome,shot_statsbomb_xg,shot_technique,shot_type,substitution_outcome,substitution_outcome_id,substitution_replacement,substitution_replacement_id,tactics,team,team_id,timestamp,type,under_pressure
0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0584ee21-e3dd-4d9f-95a0-5b5e84be25c3,1,NaN,NaN,3869685,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1,Regular Play,NaN,NaN,NaN,1,Argentina,779,NaN,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"{'formation': 433, 'lineup': [{'player': {'id'...",Argentina,779,00:00:00.000,Starting XI,NaN
1,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,b32679f8-942e-4122-96a2-015caf75e628,2,NaN,NaN,3869685,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1,Regular Play,NaN,NaN,NaN,1,Argentina,779,NaN,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,"{'formation': 4231, 'lineup': [{'player': {'id...",France,771,00:00:00.000,Starting XI,NaN
2,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,954f6855-de22-46a2-8d09-6fe94eec2b9b,3,NaN,NaN,3869685,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1,Regular Play,NaN,NaN,NaN,1,Argentina,779,[6404a8e8-afaf-489d-b65e-173a237ffed5],0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,France,771,00:00:00.000,Half Start,NaN
3,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6404a8e8-afaf-489d-b65e-173a237ffed5,4,NaN,NaN,3869685,0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,1,Regular Play,NaN,NaN,NaN,1,Argentina,779,[954f6855-de22-46a2-8d09-6fe94eec2b9b],0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Argentina,779,00:00:00.000,Half Start,NaN
4,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,e84fb5b3-270f-4427-bf97-1a6eaec7285f,1752,NaN,NaN,3869685,45,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2,From Free Kick,NaN,NaN,NaN,79,France,771,[3e8de8c9-8be8-4921-96ec-1a29ce8fd3d4],0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,France,771,00:00:00.000,Half Start,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...

In [24]:
events_finale["type"].value_counts().head(15)

type
Pass              1263
Ball Receipt*     1114
Carry              940
Pressure           361
Ball Recovery      115
Duel                98
Dribble             54
Block               50
Foul Committed      48
Clearance           45
Foul Won            44
Goal Keeper         44
Shot                38
Miscontrol          35
Dispossessed        34
Name: count, dtype: int64

In [25]:
events_finale["type"].value_counts()

type
Pass               1263
Ball Receipt*      1114
Carry               940
Pressure            361
Ball Recovery       115
Duel                 98
Dribble              54
Block                50
Foul Committed       48
Clearance            45
Foul Won             44
Goal Keeper          44
Shot                 38
Miscontrol           35
Dispossessed         34
Dribbled Past        31
Interception         28
Substitution         13
Half Start           10
Half End             10
Injury Stoppage       9
50/50                 8
Tactical Shift        7
Starting XI           2
Bad Behaviour         2
Offside               1
Shield                1
Player On             1
Player Off            1
Name: count, dtype: int64

In [26]:
os.makedirs("../data/raw", exist_ok=True)

matches.to_csv("../data/raw/matches_wc2022.csv", index=False)
france_matches.to_csv("../data/raw/france_matches.csv", index=False)
arg_matches.to_csv("../data/raw/argentina_matches.csv", index=False)
events_finale.to_csv("../data/raw/events_finale.csv", index=False)

print("Données sauvegardées ✅")

Données sauvegardées ✅
